### ISSUE TITLE
AEX-5: in which cases ping/pong needed?

### ISSUE THREAD

**Issue by davidyuk** [aeternity/AEXs#22]
Can't it be handled on transport layer?

**Comment by shekhar-shubhendu** [comment]
`ping/pong` can be handled on the transport layer but this is mainly useful for apps using `postMessage` API.

**Comment by davidyuk** [comment]
I was guessing that it is for WebSocket or so, but I can't get how it can be used in `postMessage` interface. Can you explain it?

**Comment by shekhar-shubhendu** [comment]
`postMessage` is a simple API to send/broadcast messages additionally allowing us to specify a `target` for the message. When a browser wallet extension connects to the SDK over `postMessage` we don't really know(even right after the connect message) whether the wallet is present or not until we receive the next message or the `sign` request from the SDK fails.

We can rely on the message failures or on the wallet letting `SDK` know that it disconnecting but from my point of view using `ping/pong` to detect wallet connection will provide a much better user experience as there will be no sudden changes/failures in the app and things can be handled gracefully.

**Comment by knarz** [comment]
Please, if possible, use the discussion link, if available, for issues related to specific AEXs.

**Comment by davidyuk** [comment]
@shekhar-shubhendu Are there any cases when wallet can't call `wallet.disconnect.aepp` before closing? Anyway, aepp can poll `aepp.get.address` method that shouldn't be expensive if such polling is needed in some weird case.

**Comment by davidyuk** [comment]
@shekhar-shubhendu this issue is still actual for AEX-5

**Comment by shekhar-shubhendu** [comment]
@davidyuk will have to think about it. but most probably with latest version of aex-2 this should also be removed.

**Comment by marc0olo** [comment]
no longer relevant, AEX-5 is withdrawn. I am wondering if we should propose a new standard for wallet connections. seems like this has generally been a big pain in the past
